### PR TITLE
Set `usedforsecurity=False` in hashlib methods (FIPS compliance)

### PR DESCRIPTION
**Pull request by Wauplin**
Related to https://github.com/huggingface/transformers/issues/27034 and https://github.com/huggingface/huggingface_hub/pull/1782.

**TL;DR:** `hashlib` is not a secure library for cryptography-related stuff. We are only using `hashlib` for non-security-related purposes in `diffusers` so it's fine. From Python 3.9 we can set `usedforsecurity=False` in any `hashlib` method which is mandatory for companies that forbid the use of `hashlib` for security purposes. This PR fixes that.

**Note:** before merging this we need to release a new tokenizers version that would allow the newest `huggingface_hub` version (see https://github.com/huggingface/tokenizers/pull/1385). Otherwise it might create friction to users that want to install `diffusers` + `tokenizers` at the same time.

### CODE REVIEW

**Review by sayakpaul — APPROVED** [review]
TIL!

Thanks!
